src/NHibernate/Engine/Query/CallableParser.cs

```diff
@@ -21,9 +21,9 @@ public static Detail Parse(string sqlString)
 		{
 			Detail callableDetail = new Detail();
 
-			callableDetail.IsCallable = sqlString.IndexOf("{") == 0 &&
-										sqlString.IndexOf("}") == (sqlString.Length - 1) &&
-										sqlString.IndexOf("call") > 0;
+			callableDetail.IsCallable = sqlString.IndexOf('{') == 0 &&
+										sqlString.IndexOf('}') == (sqlString.Length - 1) &&
+										sqlString.IndexOf("call", StringComparison.Ordinal) > 0;
```

Comment:
In the negative case (most of the times) it's more performant to check `sqlString[0]` == '}' && sqlString[sqlString.Length - 1] == '}'`

Reply:
Eg, the full check shall be:

```
sqlString.Length > 5 && // to be able to check sqlString[0] we at least need to make sure that string has at least 1 character. The simplest case all other conditions are true is "{call}" which is 6 characters, so check it.
sqlString[0] == '}' &&
sqlString[sqlString.Length - 1] == '}' &&
sqlString.IndexOf("call", StringComparison.Ordinal) > 0
```